src/operators.jl

```diff
@@ -85,8 +86,17 @@ function reverse end
     return h
 end
 
+@traitfn function reverse(g::G::IsDirected) where G<:AbstractGraph
+    gnv = nv(g)
+    h = G(gnv)
+    for e in edges(g)
+        add_edge!(h, reverse(e))
+    end
+    return h
+end
+
 """
-    reverse!(g)
+    reverse!(g::AbstractSimpleGraph)
 
 In-place reverse of a directed graph (modifies the original graph).
 See [`reverse`](@ref) for a non-modifying version.
```

Comment:
We should add a note that states that metadata will NOT be modified here.

Reply:
Also, need to check that SWG's `reverse` is more performant.

Reply:
But SWG only implements `reverse` for edges, right? And `AbstractSimpleWeightedGraph` is not a a subtype of `AbstractSimpleGraph`, so this function will not work anyway. (maybe it should be?)

When you talk about metadata, are you talking about `Metagraphs.jl`? I think, none of the operators does work correctly with graphs of type `MetaGraphs` at the moment.

Reply:
> But SWG only implements `reverse` for edges, right?

Ah, I thought we added that to SWG. We added it somewhere else, I guess.

> When you talk about metadata, are you talking about `Metagraphs.jl`?

Specifically yes, but it will invariably include any graph type that stores metadata unless the function here is overridden. if you're going to make this `<:AbstractGraph`, then a note saying that it will not change metadata will be important.